src/analysis.rs

```diff
@@ -96,7 +99,12 @@ impl Cost {
     /// the given transaction input.
     #[cfg(feature = "elements")]
     pub fn is_budget_valid(&self, txin: &elements::TxIn) -> bool {
-        let budget = txin.witness.script_witness.iter().map(|el| el.len()).sum();
+        let mut witness_stack_encoding = Vec::new();
+        txin.witness
+            .script_witness
+            .consensus_encode(&mut witness_stack_encoding)
+            .expect("writing to vector never fails");
```

Comment:
In a314da1acd8015e94b04e7fe9f7fd02d24f16368:

It does not seem reasonable to allocate a vector here. Why not use `Witness::serialized_len` which I even checked was available in rust-bitcoin 0.29.